workers/social/lib/social/models/computeproviders/stacktemplate.coffee

```diff
@@ -762,6 +777,41 @@ module.exports = class JStackTemplate extends Module
         Kloud.buildStack client, { stackId, provider, variables }, callback
 
 
+  verify$: permit
+
+    advanced: [
+      { permission: 'update own stack template', validateWith: Validators.own }
+      { permission: 'update stack template' }
+    ]
+
+    success: (client, callback) ->
+      @verify client, callback
+
+
+  verify: (client, callback) ->
+
+    stackTemplateId = @getAt '_id'
+    provider = (@getAt 'config.requiredProviders')[0]
+
+    unless provider or not PROVIDERS[provider]
+      return callback new KodingError 'Provider is not supported'
+
+    noMachines = ->
+      callback new KodingError \
+        'Nothing to verify, template has no machines'
+
+    Kloud = require './kloud'
+    Kloud.checkTemplate client, { stackTemplateId, provider }, (err, res) =>
+      return callback err  if err
+      return noMachines()  if not res or not res.machines?.length
+      return noMachines()  unless (machines = parsePlanResponse res).length
```

Comment:
could you please do this op `machines = parsePlanResponse res` out of the if condition, it requires back&fort to follow the code in line 809